apps/prairielearn/src/tests/groupExam.test.ts

```diff
@@ -0,0 +1,513 @@
+import { assert } from 'chai';
+import cheerio = require('cheerio');
+import fetch from 'node-fetch';
+import { config } from '../lib/config';
+import { step } from 'mocha-steps';
+
+import { queryAsync, queryOneRowAsync, loadSqlEquiv } from '@prairielearn/postgres';
+const sql = loadSqlEquiv(__filename);
+
+import helperServer = require('./helperServer');
+import { TEST_COURSE_PATH } from '../lib/paths';
+
+const siteUrl = 'http://localhost:' + config.serverPort;
+const baseUrl = siteUrl + '/pl';
+const courseInstanceUrl = baseUrl + '/course_instance/1';
+
+const storedConfig: any = {};
+
+const GROUP_NAME = 'groupBB';
+const GROUP_NAME_ALTERNATIVE = 'groupCC';
+
+interface StudentUser {
+  user_id: string;
+  uid: string;
+  name: string;
+  uin: string;
+}
+
+async function generateThreeStudentUsers(): Promise<StudentUser[]> {
+  const result = await queryAsync(sql.generate_and_enroll_3_users, []);
```

Comment:
I'd recommend using the `queryRows` function here and constructing a [Zod](https://github.com/colinhacks/zod) schema for the third argument - this ensures the query returns the data we actually expect it to! If you search for `queryRows(` in the codebase, you can find some existing usages to pattern-match against.